### PR TITLE
Fix plugin loading in manager

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR ensures new plugins are registered each time a project's dependencies are installed.

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

**Review by solarw — APPROVED** [review]
LGTM

File: aea/manager/manager.py [hunk 1]
```diff
@@ -395,6 +396,7 @@ def add_project(
 
         if not restore:
             project.install_pypi_dependencies()
+            load_all_plugins(is_raising_exception=False)
```
Comment (marcofavorito) [inline aea/manager/manager.py:399]: what happens when two AEA projects have different version, or version specifier sets, of the same plugin? `pip` should raise an error when the versions are in conflicts. When and how do we handle this?

File: aea/crypto/plugin.py [hunk 2]
```diff
@@ -157,13 +157,17 @@ def _iter_plugins() -> Iterator[Plugin]:
...
+    try:
+        register_function(plugin.name, entry_point=plugin.entry_point_path)
+    except AEAException:  # pragma: nocover
+        if is_raising_exception:
+            raise
 
 
-def load_all_plugins() -> None:
+def load_all_plugins(is_raising_exception: bool = True) -> None:
     """Load all plugins."""
```
Comment (marcofavorito) [inline aea/crypto/plugin.py:171]: please, add an explanation of what `is_rasing_exception` does.

**Review by marcofavorito — APPROVED** [review]
LGTM

**Review by panasevychol — APPROVED** [review]

**Pull request merged**